crates/biome_js_analyze/tests/specs/nursery/noYodaExpression/invalidRange.js.snap

```diff
@@ -0,0 +1,298 @@
+---
+source: crates/biome_js_analyze/tests/spec_tests.rs
+expression: invalidRange.js
+---
+# Input
+```jsx
+if (a < 0 && 0 <= b && b < 1) {}
+if (0 <= a && a < 1 && b < 1) {}
+if (1 < a && a < 0) {}
+0 < a && a < 1
+var a = b < 0 || 1 <= b;
+if (0 <= x && x < -1) {}
+if (`green` < x.y && x.y < `blue`) {}
+if (0 <= a[b()] && a[b()] < 1) {}
+if ('b' <= x && x < 'a') {}
+if('a' <= x && x < 1) {}
+if (0 < a && b < max) {}
+
+```
+
+# Diagnostics
+```
+invalidRange.js:1:14 lint/nursery/noYodaExpression  FIXABLE  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
+
+  ! Avoid the use of yoda expressions.
+  
+  > 1 │ if (a < 0 && 0 <= b && b < 1) {}
```

Comment:
Is this normal that this one is reported? We should accept ranges like `0 <= b && b < 1`?

Reply:
I've kept the same behavior of the equivalent ESLint rule, which only consider a range expression when the BinaryExpression have two assertions in that format. Probably their though on this was that `0 <= b && b < 1` **is** a range expression, but is not **just** a range expression because of the `a < 0`. 

Personally I agree with only allowing "pure" range expressions, but let me know if you want me to change this case to be valid.

Reply:
Thanks for the clarification :)